### PR TITLE
Merge v1/5-docker

### PR DESCRIPTION
**Pull request by Serphentas**

### CODE REVIEW

File: infra/dev/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - 80:80
+    volumes:
+      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
+    depends_on:
+      - laravel
+    networks:
+      - paul_net
+
+  # Here we delegate the dependency installation to composer container
+  composer:
```
Comment (kayoumido) [inline infra/dev/docker-compose.yml:16]: Composer is php dependency manager hence it should be in the php container (or laravel since you named it that way)

File: infra/dev/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    restart: always
+    ports:
+      - 5432:5432
+    volumes:
+      - ./postgres/initScript:/docker-entrypoint-initdb.d
+    environment:
+      POSTGRES_USER: laravel
+      POSTGRES_PASSWORD: passPRO
+  
+  adminer:
```
Comment (kayoumido) [inline infra/dev/docker-compose.yml]: What is the purpose of this container?

File: infra/dev/laravel/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# use Debian Buster (10) with FPM
+FROM php:7.4-fpm
+
+# unlock php deb package
+RUN rm /etc/apt/preferences.d/no-debian-php
+
+ARG USER_ID
+ARG GROUP_ID
+
+RUN groupadd -f $GROUP_ID
```
Comment (kayoumido) [inline infra/dev/laravel/Dockerfile:9]: Is this to solve the writes problems we had previously?

File: infra/dev/laravel/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# use Debian Buster (10) with FPM
```
Comment (kayoumido) [inline infra/dev/laravel/Dockerfile]: Remove all the commented stuff, it clutters the file

File: infra/dev/nginx/nginx.conf [hunk 5]
```diff
@@ -0,0 +1,37 @@
+http {
```
Comment (kayoumido) [inline infra/dev/nginx/nginx.conf:1]: Why is this needed? When I did my simple version, I just had `server { ...`

File: infra/dev/nginx/nginx.conf [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+            fastcgi_pass dev_laravel:9000;
+            fastcgi_index index.php;
+        }
+    }
+}
+
+# need an event section, even if empty
+# otherwise error on docker up
+events {
```
Comment (kayoumido) [inline infra/dev/nginx/nginx.conf:35]: Following up the other comment on this file, I didn't need this when I only had the `server { ...`

File: infra/dev/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
+version: "3.7"
+services:
+  nginx:
+    image: nginx:stable
+    container_name: dev_nginx
+    ports:
+      - 80:80
+    volumes:
```
Comment (kayoumido) [inline infra/dev/docker-compose.yml:8]: why isn't the `src` mapped for nginx?

File: infra/dev/postgres/initScript/init.sql [hunk 8]
```diff
@@ -0,0 +1,12 @@
+CREATE DATABASE laravel;
```
Comment (kayoumido) [inline infra/dev/postgres/initScript/init.sql]: This file can be deleted since db will be managed by Laravels ORM.

I'm guessing it was for testing purposes.

The Larvel command to create the db is 
```
$ php artisan migrate
```

File: infra/dev/setup-env.sh [hunk 9]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# build the container with the same UID and GID as the local user and launch the container
+USER_ID=$(id -u) GROUP_ID=$(id -g)  docker-compose up --build
```
Comment (kayoumido) [inline infra/dev/setup-env.sh:4]: What is the purpose of this `USER_ID=$(id -u) GROUP_ID=$(id -g)`? can't it be placed in the Dockerfile?

File: infra/dev/start-env.sh [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# start only the container for hosting the file
+USER_ID=$(id -u) GROUP_ID=$(id -g) docker-compose up laravel nginx db adminer
```
Comment (kayoumido) [inline infra/dev/start-env.sh:4]: Same as for `setup-env.sh` and why are you specifying the containers to up ? Isn't `$ docker-compose up` enough?

File: src/.env [hunk 11]
```diff
@@ -0,0 +1,46 @@
+APP_NAME=Laravel
```
Comment (kayoumido) [inline src/.env]: This file shouldn't be on the repo!

**Review by kayoumido — CHANGES_REQUESTED** [review]
Just check the comments I made for dev.

I'm guessing that you didn't work on prod which explains why it's still complicated

We've agreed that we won't be needing any staging and testing so you can remove them from the project

File: infra/dev/setup-env.sh (hunk 9 above)
Comment (SundayBraid) [reply infra/dev/setup-env.sh:4]: no, we need to run these command in the terminal, so I made a script for a simple use

File: infra/dev/start-env.sh (hunk 10 above)
Comment (SundayBraid) [reply infra/dev/start-env.sh:4]: same answer, no we need to run this in the terminal,these information fill the args user_id and group_id

File: infra/dev/postgres/initScript/init.sql (hunk 8 above)
Comment (SundayBraid) [reply infra/dev/postgres/initScript/init.sql]: yes

File: infra/dev/docker-compose.yml (hunk 1 above)
Comment (SundayBraid) [reply infra/dev/docker-compose.yml:16]: I separate it, since I found out there is a composer container already available (avoid the installation just for one command)

File: infra/dev/docker-compose.yml (hunk 2 above)
Comment (SundayBraid) [reply infra/dev/docker-compose.yml]: it a "php-mysql like", it can be useful to have a web UI to interrogate the db

File: infra/dev/laravel/Dockerfile (hunk 3 above)
Comment (SundayBraid) [reply infra/dev/laravel/Dockerfile:9]: these line create a user with the same level of access as the local user on the host, we found that for solving the permission problem

File: infra/dev/docker-compose.yml (hunk 2 above)
Comment (kayoumido) [reply infra/dev/docker-compose.yml]: This isn't needed, if we really need to access the db we can use tools like mysql-workbenchlike but for postgres

File: infra/dev/laravel/Dockerfile (hunk 4 above)
Comment (Serphentas) [reply infra/dev/laravel/Dockerfile]: Fixed in 4ab3fd5

File: infra/dev/nginx/nginx.conf (hunk 5 above)
Comment (Serphentas) [reply infra/dev/nginx/nginx.conf:1]: We need an empty `events` sections so a parent `server` block is therefore mandatory.

File: infra/dev/docker-compose.yml (hunk 7 above)
Comment (Serphentas) [reply infra/dev/docker-compose.yml:8]: nginx talks to Laravel through a PHP-FPM connection, on another container, and thus doesn't need to have app source onboard.

File: infra/dev/postgres/initScript/init.sql (hunk 8 above)
Comment (Serphentas) [reply infra/dev/postgres/initScript/init.sql]: Fixed in 71045c0

File: infra/dev/setup-env.sh (hunk 9 above)
Comment (Serphentas) [reply infra/dev/setup-env.sh:4]: This is a shell command, must be run from terminal.

File: src/.env (hunk 11 above)
Comment (Serphentas) [reply src/.env]: Fixed in 18fbb88

File: infra/dev/docker-compose.yml (hunk 1 above)
Comment (Serphentas) [reply infra/dev/docker-compose.yml:16]: Can't run `composer install` on Laravel as we map the app source in that container, and those files are only present during runtime and not buildtime.

File: infra/dev/docker-compose.yml (hunk 2 above)
Comment (Serphentas) [reply infra/dev/docker-compose.yml]: Fixed in 30c050b

**Review by kayoumido — APPROVED** [review]
